### DISCUSSION TITLE
Curve: weighted tangents

### DISCUSSION THREAD

**Discussion by MeisterDerMagie** [godotengine/godot-proposals#7781] [Editor]
In most curve editors out there (Unity, After Effects, Spine 2D, ...) a tangent can not only be free, but also weighted. This gives animators much more control over the "character" of a movement. Godot's lack of this feature severely limits the power of curves.

Godots curve:
![Curve_non-weighted](https://github.com/godotengine/godot-proposals/assets/40137439/5433b9e5-e1e8-4c13-9364-0bea2a2ec5d4)

Curve with weighted tangents:
![Curve_weighted](https://github.com/godotengine/godot-proposals/assets/40137439/ed5dcbd1-0b8f-4d84-a842-57b98b5f3605)

**Comment by Calinou** [comment]
I think the reason this isn't supported is that the curve could end up going "over itself", which is not allowed for Curve resources as they're 1-dimensional. Of course, clamping could be implemented to prevent this, but it might be difficult to manage when you move around other points in the curve.

In the meantime, you can use additional points to create a weighted curve.

**Reply by MeisterDerMagie** [reply]
Curves in Unity and After Effects are also 1-dimensional. You just can't pull the handle further than the previous point.

![ezgif-4-c7cf484bc3](https://github.com/godotengine/godot-proposals/assets/40137439/1eb09937-da70-40cf-8fbc-e823db7a24ed)

**Comment by MeisterDerMagie** [comment]
On second thought: this is probably not an editor thing, but the underlying curve class would also need to be reworked.

**Comment by Rouli-M** [comment]
Yes. This is much needed.

**Reply by Calinou** [reply]
@Rouli-M Please don't bump issues without contributing significant new information. Use the :+1: reaction button on the first post instead.
